src/Illuminate/Database/Eloquent/Model.php

```diff
@@ -3038,6 +3038,17 @@ public function replicate(array $except = null)
     }
 
     /**
+     * Determine if the model matches the model passed in.
+     *
+     * @param  \Illuminate\Database\Eloquent\Model  $model
+     * @return bool
+     */
+    public function is(Model $model)
+    {
+        return $this->getKey() === $model->getKey() && $this->getTable() === $model->getTable();
```

Comment:
I've only had a couple of apps that used multiple connections, so I didn't even consider that scenario. It's tough, I reckon it could go either way on this one.